ang/civicase/ActivitiesCalendar.js

```diff
@@ -66,25 +86,150 @@
+      }
+
+      /**
+       * Changes the glyphicons for the calendar arrows and replaces them with
+       * material icons.
+       */
+      function useMaterialIconsForCalendarArrows () {
+        element.find('thead .glyphicon-chevron-left')
+          .attr('class', 'material-icons')
+          .text('chevron_left');
+        element.find('thead .glyphicon-chevron-right')
+          .attr('class', 'material-icons')
+          .text('chevron_right');
+      }
     }
   });
 
   module.controller('civicaseActivitiesCalendarController', civicaseActivitiesCalendarController);
 
   function civicaseActivitiesCalendarController ($scope, formatActivity) {
-    $scope.calendarOptions = { showWeeks: false };
+    $scope.calendarOptions = {
+      customClass: getDayCustomClass,
+      formatDay: 'd',
+      showWeeks: false,
+      startingDay: 1
+    };
     $scope.selectedActivites = [];
     $scope.selectedDate = null;
 
     $scope.onDateSelected = onDateSelected;
 
+    (function init () {
+      $scope.$watch('activities', function () {
+        $scope.$broadcast('civicaseActivitiesCalendar::refreshDatepicker');
+      }, true);
+    })();
+
+    /**
+     * Determines if all the given activities have been completed.
+     *
+     * @param {Array} activities
+     */
+    function checkIfAllActivitiesHaveBeenCompleted (activities) {
+      return _.every(activities, function (activity) {
+        return _.includes(CRM.civicase.activityStatusTypes.completed, +activity.status_id);
+      });
+    }
+
+    /**
+     * Returns the activities that belong to the given date.
+     *
+     * @param {Date} date
+     */
+    function getActivitiesForDate (date) {
+      return $scope.activities.filter(function (activity) {
+        return moment(activity.activity_date_time).isSame(date, 'day');
+      });
+    }
+
+    /**
+     * Returns the class that the given date should have depending on the status
+     * of all the activities for the date.
+     *
+     * @param {Object} params
+     * @param {Date}   params.date the given date that requires the class
+     * @param {String} params.mode the current viewing mode of the calendar.
+     *   can be "day", "month", or "year".
+     */
+    function getDayCustomClass (params) {
+      var allActivitiesHaveBeenCompleted;
+      var activities = getActivitiesForDate(params.date);
+      var isDateInThePast = moment().isAfter(params.date, 'day');
+      var isInCurrentMonth = this.datepicker.activeDate.getMonth() === params.date.getMonth();
+
+      if (!isInCurrentMonth && params.mode === 'day') {
+        return 'civicase__activities-calendar__hidden-day';
+      }
+
+      if (activities.length === 0 || params.mode !== 'day') {
+        return;
+      }
+
+      allActivitiesHaveBeenCompleted = checkIfAllActivitiesHaveBeenCompleted(activities);
+
+      if (allActivitiesHaveBeenCompleted) {
+        return 'civicase__activities-calendar__day-status-completed';
```

Comment:
In BEM, variants should have `--`. 
So `civicase__activities-calendar__day-status--completed`. Please change the other two aswell.